workspaces/adoption-insights/plugins/adoption-insights/src/components/Templates/Templates.tsx

```diff
@@ -0,0 +1,164 @@
+  const entityLink = useRouteRef(entityRouteRef);
+
+  const { templates, loading } = useTemplates({ limit });
+
+  const handleChangePage = React.useCallback(
+    (_event: unknown, newPage: number) => {
+      setPage(newPage);
+    },
+    [],
+  );
+
+  const handleChangeRowsPerPage = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+      setRowsPerPage(+event.target.value);
+      setPage(0);
+    },
+    [],
+  );
+
+  const visibleTemplates = React.useMemo(() => {
+    return templates.data?.slice(
+      page * rowsPerPage,
+      page * rowsPerPage + rowsPerPage,
+    );
+  }, [templates, page, rowsPerPage]);
+
+  if (!visibleTemplates || visibleTemplates?.length === 0) {
+    return (
+      <CardWrapper title="Top templates">
+        <Box
+          display="flex"
+          justifyContent="center"
+          alignItems="center"
+          height={200}
+        >
+          <Typography>No results for this time range.</Typography>
+        </Box>
+      </CardWrapper>
+    );
+  }
+
+  return (
+    <CardWrapper title={`Top ${rowsPerPage} templates`}>
+      <Table aria-labelledby="Catalog entities" sx={{ width: '100%' }}>
+        <TableHead>
+          <TableRow>
+            {TEMPLATE_TABLE_HEADERS.map(header => (
+              <TableCell
+                key={header.id}
+                align="left"
+                sx={{
+                  borderBottom: theme => `1px solid ${theme.palette.grey[300]}`,
+                }}
+              >
+                {header.title}
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {loading ? (
+            <TableRow>
+              <TableCell colSpan={TEMPLATE_TABLE_HEADERS.length} align="center">
+                <CircularProgress />
+              </TableCell>
+            </TableRow>
+          ) : (
+            visibleTemplates?.map(template => (
+              <TableRow
+                key={template.entityRef}
+                sx={{
+                  '&:nth-of-type(odd)': { backgroundColor: 'inherit' },
+                  borderBottom: theme => `1px solid ${theme.palette.grey[300]}`,
+                }}
+              >
+                <TableCell sx={{ width: '50%' }}>
+                  <Link
+                    component="a"
+                    href={entityLink({
+                      kind: parseEntityRef(template.entityRef).kind,
+                      namespace: 'default',
+                      name: parseEntityRef(template.entityRef)?.name,
+                    })}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    sx={{
+                      textDecoration: 'none',
+                      '&:hover': {
+                        textDecoration: 'none',
+                      },
+                    }}
+                  >
+                    {parseEntityRef(template.entityRef).name}
```

Comment:
parseEntityRef function can throw error and crash the UI if the entityRef is a empty object {} or null, so it better to add safety checks before calling this function.